cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredRow.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.nosql;
+
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.data2.dataset2.lib.table.MDSKey;
+import co.cask.cdap.spi.data.InvalidFieldException;
+import co.cask.cdap.spi.data.StructuredRow;
+import co.cask.cdap.spi.data.table.StructuredTableSchema;
+import co.cask.cdap.spi.data.table.field.FieldType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * The nosql structured row represents a row in the nosql table.
+ */
+public final class NoSqlStructuredRow implements StructuredRow {
+  private final Row row;
+  private final StructuredTableSchema tableSchema;
+  private final Map<String, ? super Object> keyFields;
+
+  NoSqlStructuredRow(Row row, StructuredTableSchema tableSchema) {
+    this.row = row;
+    this.tableSchema = tableSchema;
+    this.keyFields = new HashMap<>();
+  }
+
+  @Nullable
+  @Override
+  public Integer getInteger(String fieldName) throws InvalidFieldException {
+    return get(fieldName, tableSchema.getType(fieldName));
+  }
+
+  @Nullable
+  @Override
+  public Long getLong(String fieldName) throws InvalidFieldException {
+    return get(fieldName, tableSchema.getType(fieldName));
+  }
+
+  @Nullable
+  @Override
+  public String getString(String fieldName) throws InvalidFieldException {
+    return get(fieldName, tableSchema.getType(fieldName));
+  }
+
+  @Nullable
+  @Override
+  public Float getFloat(String fieldName) throws InvalidFieldException {
+    return get(fieldName, tableSchema.getType(fieldName));
+  }
+
+  @Nullable
+  @Override
+  public Double getDouble(String fieldName) throws InvalidFieldException {
+    return get(fieldName, tableSchema.getType(fieldName));
+  }
+
+  private <T> T get(String fieldName, @Nullable FieldType.Type expectedType) throws InvalidFieldException {
```

Comment:
similar here, this method can get the field type from the schema, no need for the parameter